source/common/upstream/health_discovery_service.cc

```diff
@@ -73,5 +117,86 @@ void HdsDelegate::onRemoteClose(Grpc::Status::GrpcStatus status, const std::stri
   handleFailure();
 }
 
+HdsCluster::HdsCluster(Runtime::Loader& runtime, const envoy::api::v2::Cluster& cluster,
+                       const envoy::api::v2::core::BindConfig& bind_config, Stats::Store& stats,
+                       Ssl::ContextManager& ssl_context_manager,
+                       Secret::SecretManager& secret_manager, bool added_via_api)
+    : info_(new ClusterInfoImpl(cluster, bind_config, runtime, stats, ssl_context_manager,
+                                secret_manager, added_via_api)),
+      runtime_(runtime), initial_hosts_(new HostVector())
+
+{
+  priority_set_.getOrCreateHostSet(0);
```

Comment:
I don't think you need to maintain priority levels or localities, so no need for a priority set. You just want a single `HostSetImpl` for all the hosts in the cluster AFAICT.